### ISSUE TITLE
`diff(A, dims=1)` should make new gridpoints staggered (?)

### ISSUE THREAD

**Issue by mauro3** [rafaqz/GeoData.jl#163]
Currently it keeps the original grid, dropping the last point.

**Comment by rafaqz** [comment]
Yeah I wasn't sure what to do here, this may be wrong/a bug. 

What do you mean by staggered?

**Comment by mauro3** [comment]
Move the grid by half the grid spacing: `xnew = 0.5*(x[1:end-1] .+ x[2:end])`.  But I'm not sure whether this is "right" in all cases, it is for my current one ;-)

**Comment by rafaqz** [comment]
Im not sure that will work for DateTime. You cant e.g. shift half months generically. 

But you can hack it for your use case:

```julia
set(A, X = dims(A, X) .+ step(span(A, X))/2)
```

**Comment by mauro3** [comment]
Yes, it doesn't work for integer indices either (in general).  So not sure what's best.

**Comment by rafaqz** [comment]
Thats pretty much why it is what it is... its hard to be consistent otherwise. But Im not sure what would be "correct" in terms if the mezning of `diff`.
